controller/src/main/java/io/pravega/controller/store/stream/ZKStreamMetadataStore.java

```diff
@@ -253,6 +254,22 @@ ZKScope newScope(final String scopeName) {
         return storeHelper.checkExists(stream.getStreamPath());
     }
 
+    @Override
+    public CompletableFuture<Integer> getSafeStartingSegmentNumberFor(final String scopeName, final String streamName) {
+        return storeHelper.getData(String.format(ZKStoreHelper.DELETED_STREAMS_PATH, getScopedStreamName(scopeName, streamName)))
+                          .handleAsync((data, ex) -> {
+                              if (ex == null) {
+                                  return BitConverter.readInt(data.getData(), 0) + 1;
+                              } else if (ex instanceof StoreException.DataNotFoundException) {
+                                  return 0;
+                              } else {
+                                  log.error("Problem found while getting a safe starting segment number for {}: {}.",
```

Comment:
please remove the second {} as it is not needed. log.error will print the exception stack without it too.

Reply:
Done.